src/cpp/shared/combat/army.cpp

```diff
@@ -614,6 +650,293 @@ void army::Walk(signed int dir, int last, int notFirst) {
   }
 }
 
+int army::FindPath(int knownHex, int targHex, int speed, int flying, int flag) {
+  int res;
+  std::vector<int> obstacleHexes;
+
+  if(!IsAICombatTurn() && CreatureHasAttribute(this->creatureIdx, JUMPER) && gIronfistExtra.combat.stack.abilityCounter[this]) {
+    for (int i = 0; i < 117; i++) {
+      if (gpCombatManager->combatGrid[i].isBlocked != 0 && !IsCastleWall(i)) {
+        obstacleHexes.push_back(i);
+        gpCombatManager->combatGrid[i].isBlocked = 0;
+      }
+    }
+  }
+  res = this->FindPath_orig(this->occupiedHex, targHex, this->creature.speed, 0, flag);
+
+  // pretending we don't see obstacles at all
+  // this does nothing for creatures that don't ignore obstacles
+  for (auto i : obstacleHexes)
+    gpCombatManager->combatGrid[i].isBlocked = 1;
+
+  return res;
+}
+
+int army::ValidPath(int hex, int flag) {
```

Comment:
I can't find any difference between this and the original ValidPath. Why is this here?

Reply:
I "imported" it because -maybe- I could use it, but turned out it was not needed. I left it here because maybe it will be useful in future?

Reply:
Okay. I approve it staying.